Markdown/Markdown.sublime-syntax

```diff
@@ -180,7 +180,21 @@ variables:
     #     \s*$      # any amount of whitespace until EOL
     #   )
 contexts:
+  file-start:
+    - match: ^---$\n
+      scope: punctuation.section.block.begin.frontmatter.yaml.markdown
+      embed: scope:source.yaml
+      embed_scope: source.yaml.embedded
+      escape: ^---(?=\s)
+      escape_captures:
+        0: punctuation.section.block.end.frontmatter.yaml.markdown
+
   main:
+    - include: file-start
```

Comment:
Not sure if it is really worth adding a named context for just one match rule.

Reply:
The name of the context helps convey the semantic meaning of the match, imo, so I left that in.